sky/global_user_state.py

```diff
@@ -171,11 +180,19 @@ def update_last_use(cluster_name: str):
 
 def remove_cluster(cluster_name: str, terminate: bool):
     """Removes cluster_name mapping."""
+    handle = get_handle_from_cluster_name(cluster_name)
+    metadata = get_cluster_metadata(cluster_name)
```

Comment:
How about we directly `record = get_cluster_from_name(cluster_name)` and `handle, metadata = record['handle'], record['metadata']`?
Also, should we use the `launch_time`? It will be updated while launching on an existing cluster.